src/cargo/sources/git/oxide.rs

```diff
@@ -0,0 +1,69 @@
+//! This module contains all code sporting `gitoxide` for operations on `git` repositories and it mirrors
+//! `utils` closely for now. One day it can be renamed into `utils` once `git2` isn't required anymore.
+
+use crate::util::{network, Progress};
+use crate::{CargoResult, Config};
+use git_repository as git;
```

Comment:
It's a standard I do my best to set, as I will do this (and am doing this) wherever `git-repository` is used, consistently. If I could, I would have named `git-repository` just to be `git`, but unfortunately that's not possible on crates.io. In other words, `git-repository` as a crate considers itself to _be_ `git`.

From my experience, doing it like this works very well in practice and all names contained within `git-repository` act like they are in the `git` namespace, thus might sound wonky when used with `git_repository`.

That said, it's just a name (or namespace really), and as such that's an easy change to make.

Reply:
So you might have enough adoption at this point (or too much cost to rename everything) but having looked at this PR this helped to cement a thought that I vaguely had before: these crate names are wrong.

Ok, now that I've got that out of the way :).  At first, my thought process was as a tool author, someone taking up `git-` names that aren't git plugins was disappointing but not enough to say anything.

Here, I realized that it reads wrong in use the crates.  This isn't `git`  the tool and capturing that name, even via renames or suffixes, feels off.
- I feel like there should be a distinguishing  name to clarify which git API this is.   Purely as an analogy and not that I think there are legal issues, think of it in trademark terms.  
- `git` renames feel off as that is ambiguous as there are multiple potential `git` apis and the user to have context to know what they are looking at
- As for renames themselves, I dislike them.  It adds yet another layer of indirection for users to figure out what the code they are looking at is doing.

For me, I would evaluate renaming this, whether to `gix` (can't remember your `gix` / `ein` strategy atm to know if its taken), `gitox`, or something else.

To be clear, none of this is blocking and I understand if you don't rename them.


That also serves as a long winded way to also say that I still don't like the rename :).

Reply:
I truly hear you and took all your thoughts to heart as in "is this something that should be fixed sooner than later, and what is the right thing to do?"

> Ok, now that I've got that out of the way :). At first, my thought process was as a tool author, someone taking up git- names that aren't git plugins was disappointing but not enough to say anything.

I think that's more of a problem with a global namespace for crates with nearly no rules than anything else, with all the controversy that _unfortunately_ comes along with it. I'd feel more comfortable in `crates.io/byron/git` as well.

> As for renames themselves, I dislike them. It adds yet another layer of indirection for users to figure out what the code they are looking at is doing.

I agree, I'd like to get rid of the rename as well.

There seem to be two ways to achieve this:

1. rename `git-repository` to `gix` (name is not taken *yet*). That way it's obvious which implementation is used as well.
2. keep everything as is and hope that one day I can get the `git` crate name which was moved to `git2` and has no dependencies.
3. Do 1), solve the renaming issue, and hope for 2) nonetheless

Doing 1) or 3) would then leave all the `git-` crates inconsistent and would probably demand for a rename to `gix-*` as well which is the main reason I am not gunning for option 3) *right now*.

Option 4 might be to 'double-tap' the naming issue by releasing `git-repository` and right after, the same crate, as `gix`. Now there is ambiguity and overhead, just to avoid having to rename those `git-` crates.

With all that evaluated I feel I am back at option 2, rationalizing it as a bold move that says: there is no alternative implementations, there is `libgit2` as `git2` and there is `git`, and that's it. `git`, disguised as `git-repository` is going to be the defacto implementation and that matches with its aspiration - it's the future.
And that's probably not too far off reality unless some AI comes along and writes a production-ready complete `git` API in a week to add another option, but even then it seems fair to have the first-mover advantage, and nothing prevents that implementation to use `git-ai-*` namespaced crates if they like.
Exposing what I want quite directly: I'd love to work until it's clear that `gitoxide` deserves that `git` crate name rather than evade to something less, also saying that alternative implementations simply will have to use another name or prefix then.

I'd like to hear your thoughts on this in the hope we can find an objectively good way forward. That might be to do nothing, or at least not to rename the crate here and use `git-repository`, to avoid renaming any crate. And in case it's not clear, I am thankful to have heard your thoughts about this so openly as I truly think they expose a problem to be solved and they made me think, to help me clarify (to myself and transitively here) where I stand on this.

Reply:
> I think that's more of a problem with a global namespace for crates with nearly no rules than anything else, with all the controversy that unfortunately comes along with it. I'd feel more comfortable in crates.io/byron/git as well.

That helps with the bin case but it doesn't help inside the source code where such namespaces would disappear

> Keep everything as is and hope that one day I can get the git crate name which was moved to git2 and has no dependencies.

For me, personally, both on crates.io and in source code, I feel like if I see the name `git` it should be from the git team.  Like I said, while this isn't a legal comment I think the principle of trademark applies.

Thinking about this more, I think I can pull out one specific example of this abstract concept: you are creating an ecosystem of crates and that ecosystem needs an identifiable name.  Ideally we get [packages-as-namespaces](https://github.com/rust-lang/rfcs/pull/3243#) which helps with your crates but that doesn't help with extension crates (e.g. `git2-ext` is one of many extensions for `git2`).  If you take the `git` name, how do we make it easy to find extension crates?  `git-stack` is a bin and  `git-config-env` is a lib crate that allows you to imitate git.  Neither are gitoxide extension crates.

btw on top of `gix`, `gitox`, I will also put out there the name `gitx` ... and its taken.  nm.

Reply:
It seems the inconsistency regarding the `git` namespace is going to stay. I for one have no problem to sleep at night knowing that the `git` namespace is occupied by crates that implement it as faithfully as I can make it, while leveraging Rust every step along the way. If the `git` team ever ventures into Rust and finds a need for (one of) them, I am all ears.

Something that might help is a disclaimer in the crates README that says there is no affiliation to the c-git team. But then again, maybe that's overthinking it given that that `google-` prefixed crates I have been generating also don't carry such a disclaimer, and thus far nobody felt the need to enforce it either.

Reply:
In the course of talking about the general style I tend to use in other projects, I may not have been sufficiently clear that when contributing to Cargo I'm happy to follow whatever style the Cargo developers would prefer and will use the `git-repository` name (as of adf9c47dd1f855768caf3c9d4eee564a5f4001da). This will address the ambiguity between git2 and git in a project using both. In the course of giving some more thought to naming, `gix` would be great for this. The major reason I am so hesitant to simply rename the crates to `gix` (for `git-repository`) and `gix-*` for everything else is `cargo-smart-release` not being able to do rename tracking yet for changelog generation. But if I keep it simple, it should be reasonably straightforward to at least support renames on the same level of the tree which would be enough for this operation.

Thanks for helping me get this far in my thought process, I think I made a major breakthrough here :).